src/main/scala/com/ambiata/poacher/hdfs/Hdfs.scala

```diff
@@ -242,7 +242,10 @@ object Hdfs {
         new Path(dest, src.getName)
       else dest
     }
-    r <- if(fs.rename(src, newDest)) Hdfs.value(dest) else Hdfs.fail(s"Could not move ${src} to ${dest}")
+    parent = newDest.getParent
+    m       <- mkdir(parent)
+    _       <- if(!m) Hdfs.fail(s"'${parent}' is a file, can not move to a file!") else Hdfs.ok(())
```

Comment:
`Hdfs.fail(s"'${parent}' is a file, can not move to a file!").unless(m)`

or

`Hdfs.unless(m)(Hdfs.fail(s"'${parent}' is a file, can not move to a file!"))` /protip